graphql/cxs-impl/src/main/java/org/apache/unomi/graphql/condition/ProfileConditionFactory.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.unomi.graphql.condition;
+
+import org.apache.unomi.api.conditions.Condition;
+import org.apache.unomi.api.services.DefinitionsService;
+import org.apache.unomi.graphql.types.input.CDPInterestFilterInput;
+import org.apache.unomi.graphql.types.input.CDPProfileEventsFilterInput;
+import org.apache.unomi.graphql.types.input.CDPProfileFilterInput;
+import org.apache.unomi.graphql.types.input.CDPProfilePropertiesFilterInput;
+import org.apache.unomi.graphql.types.input.CDPSegmentFilterInput;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ProfileConditionFactory extends ConditionFactory {
+
+    public ProfileConditionFactory() {
+        super("profilePropertyCondition");
+    }
+
+    public Condition createSegmentFilterInputCondition(CDPSegmentFilterInput filterInput, Date after, Date before, DefinitionsService definitionsService) {
+        final Condition rootCondition = createBoolCondition("and", definitionsService);
+        final List<Condition> rootSubConditions = new ArrayList<>();
+
+        if (after != null) {
+            rootSubConditions.add(createDatePropertyCondition("timeStamp", "greaterThan", after, definitionsService));
+        }
+
+        if (before != null) {
+            rootSubConditions.add(createDatePropertyCondition("timeStamp", "lessThanOrEqual", before, definitionsService));
+        }
+
+        if (filterInput != null) {
+            if (filterInput.getNameEquals() != null) {
+                rootSubConditions.add(createPropertyCondition("metadata.name", filterInput.getNameEquals(), definitionsService));
+            }
+
+            if (filterInput.getViewEquals() != null) {
+                rootSubConditions.add(createPropertyCondition("metadata.scope", filterInput.getViewEquals(), definitionsService));
+            }
+
+            if (filterInput.getAndFilters() != null && filterInput.getAndFilters().size() > 0) {
```

Comment:
Please use `!isEmpty()` instead of `size() > 0` and apply that for the rest of usages